src/v2/segment.v2.ejs

```diff
@@ -0,0 +1,159 @@
+window.OptanonWrapper = getConsentAndLoadSegment;
+
+function waitForOneTrust() {
+  let attempts = 0;
+  const maxAttempts = 100;
+
+  // wait 1sec at most.
+  if (!oneTrustReady()) {
+    const interval = setInterval(() => {
+      if (oneTrustReady() || attempts > maxAttempts) clearInterval(interval);
+      attempts++;
+    }, 10);
+  }
+}
+
+function oneTrustReady() {
+    // OneTrust is ready if OnetrustActiveGroups contains at least Strictly Necessary group.
+    if ((typeof window.OnetrustActiveGroups === 'string') && (window.OnetrustActiveGroups.split(',').includes('C0001'))) {
+      return true;
+    }
+    return false;
+}
+
+async function fetchDestinations(writeKey) {
+  let destinations = await fetchDestinationForWriteKey(writeKey);
+
+  destinations = [
+    ...destinations
+      .reduce((map, item) => {
+        if (item.id === "Repeater") return map; // remove Segment Repeater destinations
+        map.has(item["id"]) || map.set(item["id"], item);
+        return map;
+      }, new Map()) // return object
+      .values()
+  ];
+
+  return destinations;
+}
+
+function getConsentAndLoadSegment() {
+  let oneTrustConsent = '';
+
+  // OneTrust stores consent in OnetrustActiveGroups.
+  oneTrustConsent = window.OnetrustActiveGroups.split(',');
+
+  const destinationPreferences = setSegmentDestinationPref(oneTrustConsent, SEGMENT_DESTINATIONS);
+
+  conditionallyLoadAnalytics({
+    writeKey: SEGMENT_WRITE_KEY,
+    destinations: SEGMENT_DESTINATIONS,
+    destinationPreferences
+  });
+}
+
+function setSegmentDestinationPref(oneTrustConsent, destinations) {
+  // maps Segment destination's category to OneTrust cookie category.
+  const segmentToOneTrust= {
+    'SMS & Push Notifications': 'C0001', // OneTrust Strictly Necessary
+    'Analytics': 'C0002', // OneTrust Performance
+    'Advertising': 'C0004' // OneTrust Targeting
+  }
+
+  const destinationPreferences = destinations
+    .map(function(dest) {
+      if ((dest.category in segmentToOneTrust) && oneTrustConsent.includes(segmentToOneTrust[dest.category])) {
+        return { [dest.id]: true };
+      } else {
+        // if we don't have a mapping for the Segment category, or we do but there's no OneTrust consent.
+        return { [dest.id]: false };
+      }
+    })
+    .reduce(
+      (acc, val) => {
+        return {
+          ...val,
+          ...acc
+        };
+      },
+      { "Segment.io": oneTrustConsent.some(d => d === SEGMENT_MAPPING) }
+    );
+
+  return destinationPreferences;
+}
+
+function conditionallyLoadAnalytics({
+  writeKey,
+  destinations,
+  destinationPreferences,
+}) {
+  let isAnythingEnabled = false;
+
+  for (const destination in destinationPreferences) {
+    const isEnabled = destinationPreferences[destination];
+    if (isEnabled) {
+      isAnythingEnabled = true;
+    }
+  }
+
+  if (isAnythingEnabled) {
+    window.analytics.load(writeKey, { integrations: destinationPreferences });
```

Comment:
On page load this call works. But when I accept all cookies via OneTrust banner, in GDPR view, it errors:
<img width="942" alt="Screen Shot 2021-11-09 at 9 36 20 AM" src="https://user-images.githubusercontent.com/63004951/140960675-061989f0-0713-4279-bc45-f03f6a129766.png">

Reply:
I think this is b/c after Segment loads, it cleans up after itself - it deletes `window.analytics.load` function. That might be why the [Gist](https://gist.github.com/sperand-io/4725e248a35d5005d68d810d8a8f7b29#file-granulardecision-js-L132) does a page reload. I took that out thinking it's not required. Now I have put it back. So now, whenever the visitor changes consent (via banner, preference-center), the page reloads. @joeyAghion , @damassi - is that okay for user experience? 
![onetrust segment test consent change reload](https://user-images.githubusercontent.com/63004951/141137200-01ef34c3-68ab-4f2a-a29e-2f6f90bf6679.gif)

Reply:
I'll leave that to Joey -- its not optimal UX and reloads are generally last resort kinda stuff (like how we do it when we log out / log in) but if the user generally only has to do it once then nbd.

Reply:
Ah, good point. Indeed, we do not yet provide user a way to recall the OneTrust banner or preference center - there's no way for user to re-consent. OneTrust saves consent in a cookie that expires in 1 year, so user will be re-prompted after a year or after clearing browser cache.

Reply:
I agree that it's not ideal, but I think it's acceptable given how we know it affects initial script execution (and how rarely these preferences should change).